cwl/src/main/scala/cwl/ExpressionEvaluator.scala

```diff
@@ -13,46 +12,31 @@ import wom.values.{WomFloat, WomInteger, WomString, WomValue}
 // http://www.commonwl.org/v1.0/CommandLineTool.html#Expressions
 object ExpressionEvaluator {
   // A code fragment wrapped in the $(...) syntax must be evaluated as a ECMAScript expression.
-  val ECMAScriptExpressionWitness = Witness("""(?s)\s*\$\((.*)\)\s*""")
+  /*
+  There are several places in cwltool where a simple contains check is used:
+  - https://github.com/common-workflow-language/cwltool/blob/353dbed/cwltool/builder.py#L157
+  - https://github.com/common-workflow-language/cwltool/blob/353dbed/cwltool/command_line_tool.py#L669
+  - https://github.com/common-workflow-language/cwltool/blob/353dbed/cwltool/expression.py#L219
+  - https://github.com/common-workflow-language/cwltool/blob/353dbed/cwltool/process.py#L645
+   */
+  val ECMAScriptExpressionWitness = Witness("""(?s).*\$\(.*""")
```

Comment:
Dare I ask what happens if that's in a quoted string?

Reply:
Thanks for the comment. I reversed (and tested) that `${return "$(";}` matches a `Function` and not an `Expression`, as the wrong order was matching the wrong type.

The copy-ported state machine can handle plain strings, so a quoted `"$("` will work just fine passing through the machine.

However, secondary files seem to handle expressions and strings differently, so with this regex match we should be effectively doing the same thing as cwltool. Now, if someone is using `$(` inside a quoted string that should not be an expression, I think they'll have a bad time in general.